pkg/kubelet/client/kubelet_client.go

```diff
@@ -93,7 +93,8 @@ func (c *KubeletClientConfig) transportConfig() *transport.Config {
 			KeyFile:  c.KeyFile,
 			KeyData:  c.KeyData,
 		},
-		BearerToken: c.BearerToken,
+		BearerToken:       c.BearerToken,
+		ConnectionTimeout: 7 * time.Second,
```

Comment:
@liggitt This timeout is the very functionality introduced by this PR. In the description I've outlined the old behaviour the introduction of this timeout seeks to revise:
>**Problem**:
>The current `client-go` code relies on the underlying (kernel) TCP stack to determine when its connection to an apiserver has dropped. When the connection is terminated gracefully this is not a problem and happens immediately, but in the case of the Kubelet becoming partitioned on the network, this can take upwards of 15 minutes. During this time, no reconnection attempts can occur…

The previous behaviour was determined by various sysctl's on Linux – more information can be found in [this comment](https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731) but is completely platform and environment specific. This PR puts this behaviour under k8s control.

Do you want me to expand on any of the explanation in the PR description? Let me know if any of it is unclear.